hazelcast/src/main/java/com/hazelcast/instance/DefaultAddressPicker.java

```diff
@@ -69,117 +71,98 @@ public void pickAddress() throws Exception {
-                 * seen this happening on Linux and Windows environments.
-                 */
-                serverSocketChannel = ServerSocketChannel.open();
-                serverSocket = serverSocketChannel.socket();
-                serverSocket.setReuseAddress(reuseAddress);
-                serverSocket.setSoTimeout(SOCKET_TIMEOUT_MILLIS);
-                try {
-                    if (bindAny) {
-                        inetSocketAddress = new InetSocketAddress(port);
-                    } else {
-                        inetSocketAddress = new InetSocketAddress(bindAddressDef.inetAddress, port);
-                    }
-                    log(Level.FINEST, "Trying to bind inet socket address:" + inetSocketAddress);
-                    serverSocket.bind(inetSocketAddress, SOCKET_BACKLOG_LENGTH);
-                    log(Level.FINEST, "Bind successful to inet socket address:" + inetSocketAddress);
-                    break;
-                } catch (Exception e) {
-                    serverSocket.close();
-                    serverSocketChannel.close();
-
-                    if (networkConfig.isPortAutoIncrement()) {
-                        port++;
-                        error = e;
-                    } else {
-                        String msg = "Port [" + port + "] is already in use and auto-increment is disabled."
-                                + " Hazelcast cannot start.";
-                        logger.severe(msg, e);
-                        throw new HazelcastException(msg, error);
-                    }
-                }
-            }
-            if (serverSocket == null || !serverSocket.isBound()) {
-                throw new HazelcastException("ServerSocket bind has failed. Hazelcast cannot start!"
-                        + " config-port: " + networkConfig.getPort() + ", latest-port: " + port, error);
-            }
-            serverSocketChannel.configureBlocking(false);
-            bindAddress = createAddress(bindAddressDef, port);
-            log(Level.INFO, "Picked " + bindAddress + ", using socket " + serverSocket + ", bind any local is " + bindAny);
-            AddressDefinition publicAddressDef = getPublicAddress(node.getConfig(), port);
+            AddressDefinition publicAddressDef = getPublicAddressByPortSearch();
             if (publicAddressDef != null) {
                 publicAddress = createAddress(publicAddressDef, publicAddressDef.port);
-                log(Level.INFO, "Using public address: " + publicAddress);
+                logger.info("Using public address: " + publicAddress);
             } else {
                 publicAddress = bindAddress;
-                log(Level.FINEST, "Using public address the same as the bind address. " + publicAddress);
+                logger.finest("Using public address the same as the bind address: " + publicAddress);
             }
-        } catch (RuntimeException re) {
```

Comment:
I removed this catch because the directly following `catch (Exception e)` should handle `RuntimeException` in the same way.